src/fs/service.go

```diff
@@ -650,3 +677,60 @@ func (service *MercuryFsService) uploadFile(writer http.ResponseWriter, request
 
 	return
 }
+
+// rename the given file path
+func renameFile(p string) string {
```

Comment:
could move all these functions related to file handling in a new go file say `file_upload_utils.go`.